libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -749,8 +749,18 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 
 			if (identifier)
 				functionDef = &functionDef->resolveVirtual(m_context.mostDerivedContract());
+			else
+			{
+				ContractType const* type = dynamic_cast<ContractType const*>(memberAccess->expression().annotation().type);
+				if (type && type->isSuper())
+				{
+					ContractDefinition const* super = type->contractDefinition().superContract(m_context.mostDerivedContract());
+					solAssert(super, "Super contract not available.");
+					functionDef = &functionDef->resolveVirtual(m_context.mostDerivedContract(), super);
```

Comment:
I wonder whether it could be avoided having to do this twice... but I guess it's not crucial.

Reply:
And it probably isn't - the expression needs to look this up, in case it's stored as function id somewhere - but we still don't want to go via the internal dispatch if it can be avoided (i.e. the function is directly called)...

But... I'm still not quite sure, though: shouldn't this still have worked before adding this part via the internal dispatch taken from the function call expression (stored there by the super member access) instead of the function definition itself?

Reply:
Before this fix here, it just took the `referencedDeclaration` property and did a direct call.

Reply:
Ah, ok. And ``referencedDeclaration`` is just the first overload in case of multiple overloads? That does sound a bit dangerous :-). But well, alright.